general_autoencoder.py

```diff
@@ -316,11 +327,31 @@ def _fit_from_processed_data(self,
                 self.Z = self.encode(self.X)
                 
             self.Xr = self.decode(self.Z)
-            self.combined_loss, self.binary_loss, self.continuous_loss, self.reg_loss = self.get_loss()
+            self.combined_loss, self.binary_loss, self.continuous_loss, self.reg_loss = self.get_loss(self.X, self.Xr)
             self.optimizer = self.optimization_method(learning_rate=self.learning_rate).minimize(self.combined_loss)
+
             if self.uses_longitudinal_data:
-                self.combined_longitudinal_loss, self.binary_longitudinal_loss, self.continuous_longitudinal_loss, self.reg_longitudinal_loss = self.get_longitudinal_loss()
-                self.longitudinal_optimizer = self.optimization_method(learning_rate=self.learning_rate).minimize(self.combined_longitudinal_loss)
+
+                # now project Z0 forward to get Z1. 
+                # This requires multiplying the age components by longitudinal_ages1 / ages
+                self.Z1 = self.get_Z1_from_Z0(self.Z)
```

Comment:
is this call signature sufficient? to get Z1, we need to know the original age + the longitudinal age.